### PR TITLE
Fix bug in quering with joint labels

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I41d63884fed05c59d03daffa0d7b7b0851a869e1

### CODE REVIEW

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 1]
```diff
@@ -4042,6 +4042,33 @@ public void testQueryVerticesByLabelsWithOneLabelNotExist() {
...
+        for (Vertex vertex : vertices) {
+            Assert.assertEquals("person", vertex.label());
+        }
+
+        vertices = g.V().hasLabel("person", "computer").hasLabel("computer")
+                    .toList();
+        Assert.assertEquals(5, vertices.size());
+        for (Vertex vertex : vertices) {
+            Assert.assertEquals("computer", vertex.label());
+        }
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java]: add case hasLabel( "computer").hasLabel("person", "computer")

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 2]
```diff
@@ -4042,6 +4042,33 @@ public void testQueryVerticesByLabelsWithOneLabelNotExist() {
...
+    public void testQueryByJointLabels() {
+        HugeGraph graph = graph();
+        init5Persons();
+        init5Computers();
+
+        GraphTraversalSource g = graph.traversal();
+
+        List<Vertex> vertices = g.V().hasLabel("person").hasLabel("computer")
+                                 .toList();
+        Assert.assertEquals(0, vertices.size());
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java]: add case hasLabel("person").hasLabel("computer")

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/optimize/TraversalUtil.java [hunk 3]
```diff
@@ -189,6 +189,7 @@ public static ConditionQuery fillConditionQuery(
             Condition condition = convHas2Condition(has, resultType, graph);
             query.query(condition);
         }
+//        query.validate();
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/optimize/TraversalUtil.java]: remove

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 4]
```diff
@@ -4042,6 +4042,33 @@ public void testQueryVerticesByLabelsWithOneLabelNotExist() {
...
+            Assert.assertEquals("person", vertex.label());
+        }
+
+        vertices = g.V().hasLabel("person", "computer").hasLabel("computer")
+                    .toList();
+        Assert.assertEquals(5, vertices.size());
+        for (Vertex vertex : vertices) {
+            Assert.assertEquals("computer", vertex.label());
+        }
+    }
```
Comment (zhoney) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java]: Add two more test cases:
g.V().hasLabel("person", "computer").hasLabel("person", "software")
g.V().hasLabel("person", "computer").hasLabel("book", "software")

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 5]
```diff
@@ -453,22 +412,22 @@ public boolean isFlattened() {
...
+            if (keys.contains(relation.key())) {
+                int keyCount = keyCounts.getOrDefault(relation.key(), 1);
+                keyCount++;
+                if (keyCount > 1) {
+                    return true;
                 }
-
+                keyCounts.put((HugeKeys) relation.key(), keyCount);
             }
+
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: remove empty line

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 6]
```diff
@@ -453,22 +412,22 @@ public boolean isFlattened() {
...
-                    }
+    public boolean hasDuplicateKeys(Set<HugeKeys> keys) {
+        Map<HugeKeys, Integer> keyCounts = new HashMap<>();
+        for (Condition condition : this.conditions()) {
+            if (!condition.isRelation()) {
+                continue;
+            }
+            Relation relation = (Relation) condition;
+            if (keys.contains(relation.key())) {
+                int keyCount = keyCounts.getOrDefault(relation.key(), 1);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: keyCounts.getOrDefault(relation.key(), 0)

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 7]
```diff
@@ -410,6 +412,25 @@ public boolean isFlattened() {
         return true;
     }
 
+    public boolean hasDuplicateKeys(Set<HugeKeys> keys) {
+        Map<HugeKeys, Integer> keyCounts = new HashMap<>();
+        for (Condition condition : this.conditions()) {
+            if (!condition.isRelation()) {
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java]: All conditions should be relations. Check them.

**Pull request merged**